src/flower-card.ts

```diff
@@ -36,6 +36,10 @@ export default class FlowerCard extends LitElement {
         this._hass = hass;
         this.stateObj = this.config?.entity ? hass.states[this.config.entity] : undefined;
 
+        if (this.config?.name) {
+            this.stateObj.attributes.friendly_name = this.config.name;
```

Comment:
I don't really like to overwrite the values in the original object, as it might cause confusion in other parts of the code or while debugging other issues.
So I'd rather do it the other way.  If config.name is not set, set it to friendly_name and then display config.name instead.

Reply:
Thanks for the feedback, I already noticed some strange behavior due to overriding the original object.  
I've changed the code to display name instead of friendly_name and use friendly_name if name is not set.

Please review the changed and cherry-pick, as the branches are conflicted now.